wandb/sdk/data_types.py

```diff
@@ -900,16 +913,70 @@ def __init__(self, data_or_path: Union[str, "TextIO"], **kwargs: str) -> None:
 
             self._set_file(tmp_path, is_tmp=True)
         elif isinstance(data_or_path, six.string_types):
-            extension = os.path.splitext(data_or_path)[1][1:]
-            if extension not in Molecule.SUPPORTED_TYPES:
-                raise ValueError(
-                    "Molecule only supports files of the type: "
-                    + ", ".join(Molecule.SUPPORTED_TYPES)
-                )
+            if os.path.isfile(data_or_path):
+                # data_or_path is the path to a file
+                extension = os.path.splitext(data_or_path)[1][1:]
+                if extension not in Molecule.SUPPORTED_TYPES:
+                    raise ValueError(
+                        "Molecule only supports files of the type: "
+                        + ", ".join(Molecule.SUPPORTED_TYPES)
+                    )
+                if extension == "mol":
+                    # rdkit mol file format
+                    self._initialize_with_rdkit(data_or_path, **kwargs)
+                else:
+                    self._set_file(data_or_path, is_tmp=False)
+            else:
+                # data_or_path is a SMILES string
+                self._initialize_with_rdkit(data_or_path, **kwargs)
+        else:
+            # data_or_path is an rdkit.Chem.rdchem.Mol object
+            self._initialize_with_rdkit(data_or_path, **kwargs)
 
-            self._set_file(data_or_path, is_tmp=False)
+    def _initialize_with_rdkit(self, data: "RDKitDataType", **kwargs) -> None:
+        rdkit_chem = util.get_module(
+            "rdkit.Chem",
```

Comment:
Can you add this requirement to the `wandb[media]` pip payload?

Reply:
Oops, sure, added.